ush/python/pygfs/task/marine_bmat.py

```diff
@@ -0,0 +1,351 @@
+        exec_name = os.path.join(self.task_config.DATA, 'gdas_socahybridweights.x')
+        exec_cmd.add_default_arg(exec_name)
+        exec_cmd.add_default_arg('soca_ensweights.yaml')
+
+        # compute the ensemble weights
+        mdau.run(exec_cmd)
+
+    @logit(logger)
+    def execute(self: Task) -> None:
+        """Generate the full B-matrix
+
+        This method will generate the full B-matrix according to the configuration.
+        """
+        chdir(self.task_config.DATA)
+        self.gridgen()                 # TODO: This should be optional in case the geometry file was staged
+        self.variance_partitioning()
+        self.horizontal_diffusion()    # TODO: Make this optional once we've converged on an acceptable set of scales
+        self.vertical_diffusion()
+        # hybrid EnVAR case
+        if self.task_config.DOHYBVAR == "YES" or self.task_config.NMEM_ENS > 2:
+            self.ensemble_perturbations()  # TODO: refactor this from the old scripts
+            self.hybrid_weight()           # TODO: refactor this from the old scripts
+
+    @logit(logger)
+    def finalize(self: Task) -> None:
+        """Finalize the global B-matrix job
+
+        This method will finalize the global B-matrix job.
+        This includes:
+        - copy the generated static, but cycle dependent background error files to the ROTDIR
+        - copy the generated YAML file from initialize to the ROTDIR
+        - keep the re-balanced ensemble perturbation files in DATAenspert
+        - ...
+
+        """
+        # Copy the soca grid if it was created
+        grid_file = os.path.join(self.task_config.DATA, 'soca_gridspec.nc')
+        if os.path.exists(grid_file):
+            logger.info(f"Copying the soca grid file to the ROTDIR")
+            FileHandler({'copy': [[grid_file,
+                                   os.path.join(self.task_config.COMOUT_OCEAN_BMATRIX, 'soca_gridspec.nc')]]}).sync()
+
+        # Copy the diffusion coefficient files to the ROTDIR
+        logger.info(f"Copying the diffusion coefficient files to the ROTDIR")
+        diffusion_coeff_list = []
+        for diff_type in ['hz', 'vt']:
+            src = os.path.join(self.task_config.DATA, f"{diff_type}_ocean.nc")
+            dest = os.path.join(self.task_config.COMOUT_OCEAN_BMATRIX,
+                                f"{self.task_config.APREFIX}{diff_type}_ocean.nc")
+            diffusion_coeff_list.append([src, dest])
+
+        src = os.path.join(self.task_config.DATA, f"hz_ice.nc")
+        dest = os.path.join(self.task_config.COMOUT_ICE_BMATRIX,
+                            f"{self.task_config.APREFIX}hz_ice.nc")
+        diffusion_coeff_list.append([src, dest])
+
+        FileHandler({'copy': diffusion_coeff_list}).sync()
+
+        # Copy diag B files to ROTDIR
+        logger.info(f"Copying diag B files to the ROTDIR")
+        diagb_list = []
+        window_end_iso = self.task_config.MARINE_WINDOW_END.strftime('%Y-%m-%dT%H:%M:%SZ')
```

Comment:
any reason why the time is the end of the window? wouldn't that be confusing since the model will restart at the beginning of the window?